README.md

```diff
@@ -8,6 +8,39 @@ This document describes the specification for developing Nextflow pipelines. Thi
 
 Input for pipelines follows the [nfcore parameters][nfcore-parameters] specification. In particular, input will be passed via a CSV file listing samples and reads (using `--input`) as described in <https://nf-co.re/viralrecon/2.6.0/usage>.
 
+# Resource requirements
+
+To define computational resource requirements for each process, we will follow the [nf-core][] standards as much as possible, where resources are adjusted by a `label` in each `process` of a pipeline. Defining these resources will be divided into these two roles:
+
+* **Pipeline developer**: Responsible for setting an appropriate `label` in each process of the NextFlow pipeline to appropriatly define required resources.
+* **System administrator**: Responsible for defining and maintaining the mapping between a `label` and the specific resources required by a pipeline.
+
+## Pipeline developer
+
+The pipeline developer will add a `label` to each `process` of a NextFlow pipeline to adjust resources. For example:
+
+```
+process name {
+    label 'process_single'
+
+    // ...
+}
+```
+
+### Accepted resource labels
+
+The following labels will be accepted:
+
+* `proccess_single`
+* `process_low`
+* `process_medium`
+* `process_high`
+* `process_very_high`: An addition over those provided by nf-core.
```

Comment:
Actually we are missing **process_high_memory** as well

Reply:
I think process_very_high replaces the need for process_high_memory. 

For process_long I would not say we need right away, but perhaps it is worth having a discussion of how the developers should convey how long a process should run. Perhaps a module specific configuration would work or have one set threshold that all processes should run within e.g. No process on azure should take more than 48 hours to complete

Reply:
If label configuration options were applied in the order defined in the config file (https://github.com/nf-core/viralrecon/blob/master/conf/base.config), you could add in multiple labels to override options and avoiding needing so many categories. For example

**pipeline.nf:**
```
process {
    label 'process_single'
    label 'process_high_memory'
}
```

**base.config:**
```
process {
    withLabel:process_single {
        cpus   = { check_max( 1                  , 'cpus'    ) }
        memory = { check_max( 6.GB * task.attempt, 'memory'  ) }
        time   = { check_max( 4.h  * task.attempt, 'time'    ) }
    }

    withLabel:process_high_memory {
        memory = { check_max( 200.GB * task.attempt, 'memory' ) }
    }
}
```

But, we had discussed this and were unsure if there was a defined order for how labels were applied. Hence, needing `process_very_high` (and possibly `process_very_long`).

We could start with the the defined resource categories in this PR, and if it turns out there is a defined order to labels, then these could be supplemented with `process_high_memory` and `process_long_time` and so on to make fine-tuned adjustments later on to resources (instead of defining more categories).